modules/serverless-typescript-context-object-reference.adoc

```diff
@@ -0,0 +1,130 @@
+[id="serverless-typescript-context-object-reference_{context}"]
+= TypeScript context object reference
+
+The `context` object has several properties that can be accessed by the function developer.
+
+[id="serverless-typescript-context-object-reference-log_{context}"]
+== log
+
+Provides a logging object that can be used to write output to the cluster logs. The log adheres to the link:https://getpino.io/#/docs/api[Pino logging API].
+
+.Example log
+[source,javascript]
+----
+function handle(context: Context) Record<string,string> {
+  context.log.info(“Processing customer”);
+  return 'OK';
+}
+----
+
+You can access the function by using the `kn func emit` command to invoke it:
+
+.Example command
+[source,terminal]
+----
```

Comment:
```suggestion
----
# When the function is running locally, use the '--sink local' flag
```
Minor suggestion, since we are not stepping through a tutorial at this point. A user looking at this reference doc may already have the function deployed.

Reply:
@lance I substituted all `--sink local` with `--sink 'http://example.function.com'` - should be intuitive enough even without a comment? If you still prefer having a comment or have other suggestion, let me know.

Reply:
@rh-max like I said, I don't consider it a blocker, but I'm not sure this is really better. `func emit` without any arguments will send a cloud event to the function in the running cluster - no need to specify a URL. `func emit --sink local` will send the event to `https://localhost:8080`, which presumably is the locally running function. And finally, `func emit --sink https://example.function.com` will send a cloud event to the URL provided. Given all of that, if you think `--sink http://example.function.com` is intuitive enough, no worries. Just want to make sure it's clear.

Reply:
@lance Thanks for this. I see your point, but unfortunately I don't see how to improve this. The explanations of the various sinks are explained in [1] and are off-topic here, such explanations should be offloaded to [1]. But unfortunately we are still not allowed to link to another part of the guide from an arbitrary place in text :( so this example remains somewhat underexplained. We'll fix such problems once we are able to link from anywhere.
[1] https://deploy-preview-35587--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/kn-func-ref.html#serverless-kn-func-emit_kn-func-ref